notebooks/cloudsat_monthly_means.ipynb

```diff
@@ -0,0 +1,313 @@
+    "    day1 = 1\n",
+    "    for month in range(1, 13):\n",
+    "\n",
+    "        # Get ordinal days for the current month\n",
+    "        ndays = calendar.monthrange(year, month)[1]\n",
+    "        ordinal_days = np.arange(day1, day1+ndays)\n",
+    "        day1 += ndays\n",
+    "\n",
+    "        # cumulative weighted sums\n",
+    "        cumulative_wsums = {variable : 0.0 for variable in variables}\n",
+    "        # cumulative sum of weights\n",
+    "        #cumulative_w = {variable : 0.0 for variable in variables}\n",
+    "        cumulative_w = 0.0\n",
+    "\n",
+    "        for day in ordinal_days:\n",
+    "            \n",
+    "            # get all files for the current day\n",
+    "            date = f'{year:04}{day:03}'\n",
+    "            files = fnmatch.filter(\n",
+    "                os.listdir(input_dir), \n",
+    "                f'ccicgridsat_dardar_2cice_{date}*'\n",
+    "            )\n",
+    "            for i, file in enumerate(files):\n",
+    "                print(f'{year}-{str(month).zfill(2)}-{str(day).zfill(3)} file {i+1}/{len(files)}: {file}')\n",
+    "                # load data \n",
+    "                filepath = os.path.join(input_dir, file)\n",
+    "                with xr.open_dataset(filepath) as ds:\n",
+    "                    # remove cases where Cloudsat is based on only Lidar\n",
+    "                    idxs = ~np.isnan(ds.cloud_mask_2cice.values)\n",
+    "                    ds = ds.isel(rays=idxs)\n",
+    "                    # apply geographical mask\n",
+    "                    cloudsat_coords = np.stack(\n",
+    "                        [ds.latitude.values, ds.longitude.values],\n",
+    "                        axis=1\n",
+    "                    )\n",
+    "                    ray_mask = interpn(\n",
+    "                        (mask.latitude.values, mask.longitude.values),\n",
+    "                        mask.values.astype(int),\n",
+    "                        cloudsat_coords,\n",
+    "                        method='nearest',\n",
+    "                        bounds_error=False,\n",
+    "                        fill_value=0,\n",
+    "                    )\n",
+    "                    ray_idxs = ray_mask == 1\n",
+    "                    ds = ds.isel(rays=ray_idxs)\n",
+    "                    # remove nans\n",
+    "                    for variable in variables:\n",
+    "                        idxs = ~np.isnan(ds[variable].values)\n",
+    "                        ds = ds.isel(rays=idxs)\n",
+    "\n",
+    "                    # get weights \n",
+    "                    w = np.cos(ds.latitude.values*(np.pi/180))\n",
+    "                    cumulative_w += np.sum(w)\n",
+    "                    # calculate sums\n",
+    "                    for variable in variables:\n",
+    "                        if variable.startswith('cloud_mask'):\n",
+    "                            # make it binary baby\n",
```

Comment:
Hahahahaha, this gave me a good laugh
If I have to be picky, this doesn't enforce it to make it binary. `cloud_mask` can still be negative (comes from the data_processing.py module). But I have not idea if this happens in practice and, anyhow, can be dealt with a posteriori and it's good to differentiate between 0 (no cloud), 1 (cloud), and negative (something's wrong). In any case, since we discarded the cloud mask analysis at this point, this is somewhat irrelevant.